entity/store.go

```diff
@@ -259,7 +259,7 @@ func (s *store) ReadEntities(entityType string, q query.Query) ([]etre.Entity, e
 
 	mgoQuery := translateQuery(q)
 
-	var entities []etre.Entity
+	entities := []etre.Entity{}
```

Comment:
I think the original was better because `var entities []etre.Entity` makes no allocation, whereas `entities := []etre.Entity{}` does allocate (an empty slice). If the rare case that the query fails and doesn't use the var, we've not wasted an allocation. A micro-optimization for sure, but an easy one.

Reply:
you'd revert it for the ReadEntities as well? it felt weird for the API to return nil instead of an empty json struct if a query returns no entities

Reply:
I think it will be allocated to an empty slice when no entities match? So if `Find` funs but matches nothing, we still get a non-nil slice.  In any case, it's no concern so either way is fine.